### ISSUE TITLE
Different SDF Samples than DeepSDF. 

### ISSUE THREAD

**Issue by philippwulff** [marian42/mesh_to_sdf#41]
Hi, I looked into this library as an alternative for the mesh2sdf preprocessing pipeline in DeepSDF and I noticed that this library chooses the SDF sample location much closer to the boundary than the algorithm in DeepSDF. Why is that? Is it deliberate and does't this have negative results on training DeepSDF style models, since the sampling locations x,y,z are smaller values and therefore more difficult to learn for a network?

Let me show an example:

### Loading a preprocessed shape from DeepSDF and converting a mesh to an SDF via this library:
```
deepsdf_sdf = "data/SdfSamples/ShapeNetV2/04256520/1a4a8592046253ab5ff61a3a2a0e2484.npz"
deepsdf_mesh = "/mnt/hdd/ShapeNetCore.v2/04256520/1a4a8592046253ab5ff61a3a2a0e2484/models/model_normalized.obj"

samples = np.load(deepsdf_sdf)
neg_samples = samples["neg"]
pos_samples = samples["pos"]
points_dsdf = np.concatenate((neg_samples[:,:3], pos_samples[:,:3]), axis=0)
sdf_dsdf = np.concatenate((neg_samples[:,3], pos_samples[:,3]), axis=0)

mesh = trimesh.load(deepsdf_mesh)
points_py, sdf_py = sample_sdf_near_surface(mesh, number_of_points=500000)

# plotting the SDF values
plt.hist(sdf_dsdf, bins=100, label="DeepSDF")
plt.hist(sdf_py, bins=100, label="mesh_to_sdf")
plt.legend();
```

![Screenshot 2023-01-10 at 23 48 09](https://user-images.githubusercontent.com/45896065/211678695-f67a4b1b-9e50-4253-be72-f3bcee1e4b08.jpg)

As you can see, the SDF values given by this library are much closer to zero than with the DeepSDF tool.

Rendering the SDF from `mesh_to_sdf`:
![Screenshot 2023-01-10 at 23 48 55](https://user-images.githubusercontent.com/45896065/211678789-230bf7f8-f354-4ff5-b00d-1e49bee36cdc.jpg)

Rendering the SDF from DeepSDF's preprocessing pipeline:
![Screenshot 2023-01-10 at 23 50 00](https://user-images.githubusercontent.com/45896065/211678920-ba0aabc1-aac2-45c3-a4b2-5f5744c6b52e.jpg)

I looked at the for choosing the sampling locations in this library and in the DeepSDF code and it looked almost identical. I only think, that `mesh_to_sdf` uses a twice as large variance (even though this difference seems to require much than just `x2`). I also looked at other examples and what I found is not unique.

**Comment by philippwulff** [comment]
Update: Scaling the standard deviation in your code used for selecting the sample positions close to the surface by 10 gives me this result:

![Screenshot 2023-01-11 at 00 04 20](https://user-images.githubusercontent.com/45896065/211680709-0dc1a56e-be9b-42a0-b125-212ea7be08cb.jpg)

which is closer to DeepSDF.

**Comment by alvinsunyixiao** [comment]
DeepSDF specifies the noise **variance** while the parameter in this repo is specified using **standard deviation**. You should take the square root to compare.

**Comment by dqj5182** [comment]
@philippwulff May I ask how you scaled the standard deviation in the code?

**Comment by philippwulff** [comment]
@dqj5182 It's been a while, but I remember that there Is a line pretty deep down in the source code that you need to modify for that.

**Comment by dqj5182** [comment]
@philippwulff Thanks for the reply! Would the code you mentioned be this part(https://github.com/marian42/mesh_to_sdf/blob/66036a747e82e7129f6afc74c5325d676a322114/mesh_to_sdf/surface_point_cloud.py#L59)?

**Comment by philippwulff** [comment]
No, you need to find the code that samples locations. I couldn’t find it just now, but it is there somewhere.

**Comment by JiaoZixun** [comment]
Have you found the part that affects the sampling position? I have tested some places, and the sampling point is still close to 0
